### PR TITLE
[API-1629]: Add per-environment config

### PR DESCRIPTION
**Pull request by therockstorm**
This allows you to set credentials for multiple environments. Run `vertex configure -b [BASE_PATH]` for each environment. Follow-on commands will then work in any environment without having to swap environment variables.

If you have `VERTEX_CLIENT_ID` and `VERTEX_CLIENT_SECRET` exported, however, they will take precedence.

### CODE REVIEW

File: src/commands/configure.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      default: client?.id,
+      required: false,
+    });
+    const secret = await cli.prompt(`Vertex client secret`, {
+      default: client?.secret,
+      required: false,
+      type: 'hide',
+    });
+
+    const configPath = join(this.config.configDir, 'config.json');
```
Comment (jareddellitt) [inline src/commands/configure.ts:32]: Thoughts on using a `.env` file? Then you wouldn't need to manage the json schema and the `.env` file could be used for other purposes as well.

https://github.com/motdotla/dotenv#readme

Comment (therockstorm) [reply src/commands/configure.ts:32]: Where would this file live on customer machines? Home directory? How would we distinguish between different environments?

Comment (jareddellitt) [reply src/commands/configure.ts:32]: I don't know. Just trying to avoid implementing our own configuration management system.

**Review by jareddellitt — APPROVED** [review]

File: src/commands/configure.ts (hunk 1 above)
Comment (therockstorm) [reply src/commands/configure.ts:32]: I'm using Oclif's approach to config, https://oclif.io/docs/config#docsNav

**Review by danschultz — APPROVED** [review]

**Pull request merged**